### PR TITLE
[Docs] Add info about test proxy sanitizer opt-out

### PR DESCRIPTION
**Pull request by pvaneck**
Added some info and code snippets regarding test proxy sanitization opt-outs.

### CODE REVIEW

**Review by mccoyp — COMMENTED** [review]
Thank you so much! It'll be super helpful to be able to point to this documentation.

File: doc/dev/test_proxy_troubleshooting.md [hunk 1]
```diff
@@ -144,6 +144,28 @@ To match requests for query parameter content instead of exact ordering, you can
 `ignore_query_ordering=True`. Calling this method inside the body of a test function will update the matcher for only
 that test, which is recommended.
 
+### Sanitization impacting URL structure
+
+In some cases, values in response bodies are used to construct URLs for subsequent requests or are used as request URLS themselves. If these values are sanitized, the recorded request URL might differ than what is expected during playback. Common culprits include sanitization of "name", "id", and "Location" fields. To resolve this, you can opt out of sanitization for the fields that are used for your request URLs by calling the `remove_batch_sanitizer` method from `devtools_testutils` with the [sanitizer IDs][test_proxy_sanitizers] to exclude. Generally, this is done in the `conftest.py` file, in the one of the session-scoped fixtures. Example:
```
[the remark this answers is not in the document]
Comment (pvaneck) [reply doc/dev/test_proxy_troubleshooting.md:149]: Ah, good suggestion! I will also outline that approach as well.

File: doc/dev/tests.md [hunk 2]
```diff
@@ -576,6 +593,7 @@ For information about more advanced testing scenarios, refer to the [advanced te
 [pytest_logging]: https://docs.pytest.org/en/stable/logging.html
 [python-dotenv_readme]: https://github.com/theskumar/python-dotenv
 [recording_move]: https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/recording_migration_guide.md
+[test_proxy_sanitizers]: https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/SanitizerDictionary.cs#L65
```
Comment (mccoyp) [inline doc/dev/tests.md:596]: Same comment about pointing to a consistent commit:
```suggestion
[test_proxy_sanitizers]: https://github.com/Azure/azure-sdk-tools/blob/57382d5dc00b10a2f9cfd597293eeee0c2dbd8fd/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/SanitizerDictionary.cs#L65
```

**Review by mccoyp — APPROVED** [review]

File: doc/dev/tests.md (hunk 2 above)
Comment (pvaneck) [reply doc/dev/tests.md:596]: Thanks!

**Pull request merged**